src/FuelClient.cc

```diff
@@ -550,7 +550,12 @@ Result FuelClient::DeleteUrl(const ignition::common::URI &_uri,
 //////////////////////////////////////////////////
 Result FuelClient::DownloadModel(const ModelIdentifier &_id)
 {
-  return this->DownloadModel(_id, {});
+  std::vector<std::string> headers;
+  if (!_id.Server().ApiKey().empty())
+  {
+    headers.push_back("Private-token: " + _id.Server().ApiKey());
+  }
+  return this->DownloadModel(_id, headers);
```

Comment:
I think there is a small amount of danger with this approach. Right now the `--header` command line option is used to specify access tokens. However, it could be used for other headers in the future. So, we could end up with a command like `ign fuel download ... --headers "SOME_OTHER_HEADER"`. In this case the token specified in the config file will not be used.

You could add logic to the `DownloadModel(const ModelIdentifier &_id,
    const std::vector<std::string> &_headers)` function that checks the contents of `_headers` for `Private-token`, and then add it when missing.

Reply:
I added the private-token header in the two cases that the headers were empty

## FuelClient::DownloadModel(const ModelIdentifier &_id) 
 - https://github.com/ignitionrobotics/ign-fuel-tools/blob/c0ec7e960fa87ba90690774bebd478acaaaff7bc/src/FuelClient.cc#L551-L554
 - https://github.com/ignitionrobotics/ign-fuel-tools/blob/6b2cb96962e62b269e040ffaf67ae369991a6c24/src/FuelClient.cc#L551-L559
 
## Result FuelClient::DownloadWorld(WorldIdentifier &_id)

- https://github.com/ignitionrobotics/ign-fuel-tools/blob/c0ec7e960fa87ba90690774bebd478acaaaff7bc/src/FuelClient.cc#L687-L691 
- https://github.com/ignitionrobotics/ign-fuel-tools/blob/6b2cb96962e62b269e040ffaf67ae369991a6c24/src/FuelClient.cc#L692-L702 


This changes should not affect this method: `DownloadModel(const ModelIdentifier &_id, const std::vector<std::string> &_headers)`

Reply:
I think the problem is with the command line option `--headers` and that it is not a mechanism tied only to authentication. The `--headers` is meant to support any HTTP header.

So, a user could run `ign fuel download --headers "key: value"`. A `Private Token` contained in the config file would not be used, and this would be an unexpected result.